hera_cal/lst_stack/calibration.py

```diff
@@ -132,240 +174,350 @@ def lstbin_absolute_calibration(
                 ~stack.flags[:, apidx, :, polidx]
             ).astype(float)
 
-    # Check to see if all nights are flagged
-    all_nights_flagged = len(data_here) == 0
+    # If autos are provided and use_autos_for_abscal is True, use them for amplitude calibration
+    if use_autos_for_abscal and auto_stack is not None:
+        for polidx, pol in enumerate(stack.pols):
+            for apidx, (ant1, ant2) in enumerate(auto_stack.antpairs):
+
+                blpol = (ant1, ant2, pol)
+
+                # Move to the next blpol if there is not a model for the data or the entire baseline is flagged
+                if np.all(auto_stack.flags[:, apidx, :, polidx]):
+                    continue
+
+                # Get model, weights, and data for each baseline
+                abscal_model[blpol] = auto_model[apidx, :, polidx] * np.ones(
+                    (len(auto_stack.nights), 1)
+                )
+                data_here[blpol] = auto_stack.data[:, apidx, :, polidx]
+                wgts_here[blpol] = auto_stack.nsamples[:, apidx, :, polidx] * (
+                    ~auto_stack.flags[:, apidx, :, polidx]
+                ).astype(float)
 
     # Perform amplitude calibration
-    if run_amplitude_cal and not all_nights_flagged:
-        # Store gain solutions in paramter dictionary
-        solution = abscal.abs_amp_lincal(
-            model=abscal_model, data=data_here, wgts=wgts_here, verbose=False
+    solution = abscal.abs_amp_lincal(
+        model=abscal_model, data=data_here, wgts=wgts_here, verbose=False
+    )
+
+    calibration_parameters = {}
+
+    # Get the unique antenna-polarizations
+    unique_pols = list(
+        set(sum(map(list, [utils.split_pol(pol) for pol in stack.pols]), []))
+    )
+
+    for pol in unique_pols:
+        # Calibration parameters store in an N_nights by N_freqs array
+        amplitude_gain = np.where(
+            np.isfinite(solution[f"A_{pol}"]), solution[f"A_{pol}"], 1.0 + 0.0j
```

Comment:
Do we want to set this to 1.0 + 0.0j here, or rely on passing a nan/inf up to the higher level function where a decision can be made as to what to do with it? 

I'd be more comfortable if there was a check that all the weights are 0 rathert han if the check was just on whether `solution` were finite

Reply:
Or maybe just use the flags directly?

Reply:
I think I'll just use the flags to be safe.